runtime/near-vm-logic/src/logic.rs

```diff
@@ -13,6 +13,105 @@ use std::mem::size_of;
 
 type Result<T> = ::std::result::Result<T, HostErrorOrStorageError>;
 
+pub struct GasCounter<'a> {
```

Comment:
We could put `GasCounter` into is separate module `gas_counter.rs` and keep it under `logic` module which we can promote to a directory `logic/mod.rs` to facilitate encapsulation. Then we could add gas counting-specific unit tests to `gas_counter.rs`. 

I think this refactoring is a good idea in general because gas counting is so special compared to the rest of the logic:
* We need to have special tests for it;
* We will be doing a lot of performance optimization of its operations.

So it makes sense to keep it separate.

Reply:
Created an issue
https://github.com/nearprotocol/nearcore/issues/1394